### DISCUSSION TITLE
UPS via NUT USB/ENET Support

### DISCUSSION THREAD

**Discussion by STGMavrick** [theyosh/TerrariumPI#826] [Ideas]
Hey! I've been using it for about a year. Have you any thoughts about NUT integration in the UI for running a terrarium on a UPS?

**Comment by theyosh** [comment]
Hmm, I am afraid that will not be happening. This is in my opinion outside the range of my software.

A UPS should be handled by the OS, not by a single program. So when power is lost, and the UPS is almost empty, it should tell the OS to shutdown. And that will also shutdown TerrariumPI.

And it is very hard to make software integration when I do not have the hardware to test with. And I am not planning of buying a UPS.

Just wondering, does it happen often that you loose power? I do not have that problem.

**Comment by STGMavrick** [comment]
Power outages does happen from time to time here but not a normal occurrence but since they can fry devices I thought it worth recommending + early warning for those that aren't home and have habitats that are sensitive to light/humidity changes, etc. NUT server itself is a service installed at the OS level that manages any and all comms between all the supported UPS models, however the client portion works as a "hey, this is the info i have available for you to monitor". I haven't done written language programming in almost 20 years (I ended up going down the PLC programming path) so I'd be a bad judge of how easy it is to implement. Just thought i'd mention it as a feature in case someone hadn't thought of it yet and could see a benefit to it. No worries, if there's no interest though!

https://networkupstools.org/docs/developer-guide.chunked/ar01s08.html
